### PR TITLE
New Feature: Addition of Document insertion method for Knowledge Graph

### PR DESCRIPTION
**Pull request by Rana-Banerjee**
# Description

New Feature: Addition of Document insertion method for Knowledge Graph Index.
Exisitng _insert method inserts only nodes and not documents. I have updated the docstring of _insert method as well, stating the same.
Additional processing is required for documents to be uploaded to the same index, which is done by the new insert method.
Added necessary imports as well.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-core/llama_index/core/indices/knowledge_graph/base.py [hunk 1]
```diff
@@ -240,6 +241,19 @@ def _insert(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
...
                     self._index_struct.add_to_embedding_dict(triplet_str, rel_embedding)
 
+    def insert(self, document: Document, **insert_kwargs: Any) -> None:
+        """Insert a document."""
+        with self._service_context.callback_manager.as_trace("insert_document"):
+            for doc in document:
+                self._docstore.set_document_hash(doc.get_doc_id(), doc.hash)
+            nodes = run_transformations(
+                document,
+                self._service_context.transformations,
```
Comment (nerdai) [inline llama-index-core/llama_index/core/indices/knowledge_graph/base.py:251]: `ServiceContext` is deprecated. New features/capabilities should refrain from using it. Moreover, these shouldn't these transformation be stored already in `self._transformations`?

File: llama-index-core/llama_index/core/indices/knowledge_graph/base.py [hunk 2]
```diff
@@ -240,6 +241,19 @@ def _insert(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
                     rel_embedding = self._embed_model.get_text_embedding(triplet_str)
                     self._index_struct.add_to_embedding_dict(triplet_str, rel_embedding)
 
+    def insert(self, document: Document, **insert_kwargs: Any) -> None:
```
Comment (nerdai) [inline llama-index-core/llama_index/core/indices/knowledge_graph/base.py:244]: We have `insert_nodes` already that exists on the `BaseIndex` class. If we do implement this, then we should use `insert_documents` or `insert_nodes_from_documents` instead.

**Review by nerdai — COMMENTED** [review]
Thanks for this @Rana-Banerjee. I certainly get why some of our users would want this type of interface to add nodes to an existing index.

If we do this however, I think its better defined on the `BaseIndex` and thus available to all subclasses including `KnowledgeGraphIndex`

**Comment by nerdai** [comment]
> Thanks for this @Rana-Banerjee. I certainly get why some of our users would want this type of interface to add nodes to an existing index.
> 
> If we do this however, I think its better defined on the `BaseIndex` and thus available to all subclasses including `KnowledgeGraphIndex`

@logan-markewich: Thoughts?

File: llama-index-core/llama_index/core/indices/knowledge_graph/base.py (hunk 2 above)
Comment (logan-markewich) [reply llama-index-core/llama_index/core/indices/knowledge_graph/base.py:244]: @nerdai there's an insert() method on the base class already right? (Although it takes a single document, not a list, for whatever reason)

Comment (nerdai) [reply llama-index-core/llama_index/core/indices/knowledge_graph/base.py:244]: ah yes! how did i miss that -- my bad. So in this case, this should already be available to KnowledgeGraphIndex

**Comment by Rana-Banerjee** [comment]
@logan-markewich @nerdai The base index does cover the functionality. It was a miss on my part. Thanks for pointing it out. Closing the request.

**Pull request closed without merging**